### PR TITLE
display health policy

### LINKED ISSUE #714: Show more health check duration info during upgrades
**Is your feature request related to a problem? Please describe.**
Really like the improvements to cluster repair tasks showing the breakdown and duration for each phase. However, for app upgrades, that information is still missing.

**Describe the solution you'd like**
Show the time spent (and remaining) for app upgrade health checks. Right now the UI doesn't give any indication of progress.

This could even be a % process bar based on time remaining, right under "Time left until timeout" - could say something like "Time left until health check stable duration completes" (which would reflect how long the app health has been stable, resetting appropriately if app becomes unhealthy)

**Additional context**
Would like more visual feedback on % time remaining for health checks:
image

### PR DESCRIPTION
**Pull request by jeffj6123**
DIsplay the wait/stable/retry timers during an upgrade

closes #714 
tooltip

### CODE REVIEW

File: src/SfxWeb/src/app/modules/upgrade-progress/upgrade-info/upgrade-info.component.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { ClusterUpgradeProgress } from 'src/app/Models/DataModels/Cluster';
 import { ListSettings } from 'src/app/Models/ListSettings';
 import { IEssentialListItem } from 'src/app/modules/charts/essential-health-tile/essential-health-tile.component';
 import { SettingsService } from 'src/app/services/settings.service';
+import { IProgressStatus } from 'src/app/shared/component/phase-diagram/phase-diagram.component';
```
Comment (chensation) [inline src/SfxWeb/src/app/modules/upgrade-progress/upgrade-info/upgrade-info.component.ts:7]: unused import

File: src/SfxWeb/src/app/shared/component/display-duration/display-duration.component.ts [hunk 2]
```diff
@@ -16,12 +21,20 @@ export class DisplayDurationComponent implements OnChanges {
   @Input() bottomHelpText = '';
   @Input() bottomHelpTextLink = '';
 
+  @Input() colorMap: Record<number, string> = {
+    .75 : 'red',
+    .5 : 'yellow',
+    0 : 'var(--accent-darkblue)'
```
Comment (chensation) [inline src/SfxWeb/src/app/shared/component/display-duration/display-duration.component.ts:27]: _vars.scss has `--badge-warning` and `--badge-error` for red and yellow

File: src/SfxWeb/src/app/shared/component/display-duration/display-duration.component.ts [hunk 3]
```diff
@@ -30,16 +43,30 @@ export class DisplayDurationComponent implements OnChanges {
...
     this.barColor = this.setColorCode(this.leftDuration);
+
+    this.sections = [
+      {
+        width: this.leftDuration,
+        color: this.barColor
+      },
+      {
+        width: this.rightDuration,
+        color: "grey"
```
Comment (chensation) [inline src/SfxWeb/src/app/shared/component/display-duration/display-duration.component.ts:54]: similarly, can we use one of the colors in _vars.scss, such as `--disabled`

File: src/SfxWeb/src/app/modules/upgrade-progress/upgrade-info/upgrade-info.component.scss [hunk 4]
```diff
@@ -2,9 +2,12 @@
     margin-top: 5px;
 }
 
-h3 {
+h3, h4 {
```
Comment (chensation) [inline src/SfxWeb/src/app/modules/upgrade-progress/upgrade-info/upgrade-info.component.scss:5]: where are h4 elements?

File: src/SfxWeb/src/app/modules/upgrade-progress/health-policy-check/health-policy-check.component.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+      this.color = 'var(--badge-error)';
+      this.failed = true;
+    } else if (this.healthCheckPhase === "StableDuration") {
+      this.currentPhaseIndex = 2;
+      minDurationLeft = TimeUtils.getDuration(TimeUtils.getDurationMilliseconds(this.monitoringPolicy.HealthCheckStableDurationInMilliseconds) - this.healthCheckDurationLeft);
+      this.healthCheckPhaseText = "If the health policy becomes unhealthy, the stable check will move to retry and potentially fail.";
+      this.HealthCheckDurationOverall = TimeUtils.getDurationMilliseconds(this.monitoringPolicy.HealthCheckStableDurationInMilliseconds);
+      this.displayBottomText = "Stable Time Duration Left";
+      this.displayTopText = "Stable Time Elapsed";
+      this.color = 'green';
```
Comment (chensation) [inline src/SfxWeb/src/app/modules/upgrade-progress/health-policy-check/health-policy-check.component.ts:81]: any reason not to use `--badge-ok` for green here?

**Review by chensation — CHANGES_REQUESTED** [review]

File: src/SfxWeb/src/app/modules/upgrade-progress/upgrade-info/upgrade-info.component.scss (hunk 4 above)
Comment (jeffj6123) [reply src/SfxWeb/src/app/modules/upgrade-progress/upgrade-info/upgrade-info.component.scss]: removed reference.

File: src/SfxWeb/src/app/modules/upgrade-progress/health-policy-check/health-policy-check.component.ts (hunk 5 above)
Comment (jeffj6123) [reply src/SfxWeb/src/app/modules/upgrade-progress/health-policy-check/health-policy-check.component.ts]: added css var to start moving to "proper" green

**Review by chensation — APPROVED** [review]

**Pull request merged**